cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/engine/SQLEngine.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.etl.api.engine;
+
+import io.cdap.cdap.api.annotation.Beta;
+import io.cdap.cdap.api.data.batch.InputFormatProvider;
+import io.cdap.cdap.api.data.batch.OutputFormatProvider;
+import io.cdap.cdap.etl.api.PipelineConfigurable;
+import io.cdap.cdap.etl.api.SubmitterLifecycle;
+import io.cdap.cdap.etl.api.batch.BatchContext;
+import io.cdap.cdap.etl.api.join.JoinDefinition;
+
+/**
+ * A SQL Engine can be used to pushdown certain dataset operations.
+ *
+ * SQL Engines are implemented as plugins.
+ */
+@Beta
+public interface SQLEngine extends PipelineConfigurable, SubmitterLifecycle<BatchContext> {
+  String PLUGIN_TYPE = "sqlengine";
+
+  /**
+   * Creates an Output Format Provided that can be used to push records into a SQL Engine.
+   * @param tableName The name of the table to use to store these records.
+   * @return an {@link OutputFormatProvider} instance that can be used to write records to the SQL Engine into the
+   * specified table.
+   */
+  OutputFormatProvider push(String tableName);
```

Comment:
nit: 'push' sounds like the data would be written by this method. It may be better to name it something like getOutputFormatProvider(), or getPushFormat()